bitcoin/src/blockdata/script.rs

```diff
@@ -673,21 +673,11 @@ impl fmt::UpperHex for Script {
     }
 }
 
-impl hex::FromHex for ScriptBuf {
-    #[inline]
-    fn from_byte_iter<I>(iter: I) -> Result<Self, hex::Error>
-    where
-        I: Iterator<Item=Result<u8, hex::Error>> + ExactSizeIterator + DoubleEndedIterator,
-    {
-        Vec::from_byte_iter(iter).map(ScriptBuf)
-    }
-}
-
 impl core::str::FromStr for ScriptBuf {
     type Err = hex::Error;
     #[inline]
     fn from_str(s: &str) -> Result<Self, hex::Error> {
-        hex::FromHex::from_hex(s)
+        ScriptBuf::from_hex(s)
```

Comment:
Oh, we had `FromStr` despite it not being inverse of `Display`. That's not great, I think we should remove it. (We could also parse human-friendly script but do we want to?)